### PR TITLE
Adapt frontend/backend for rename_tag request

### PR DESCRIPTION
**Pull request by zaucker**

### CODE REVIEW

File: lib/Agrammon/Web/Routes.pm6 [hunk 1]
```diff
@@ -165,11 +165,11 @@ sub dataset-routes(Agrammon::Web::Service $ws) {
...
 
-        # test
+        # working
         post -> LoggedIn $user, 'rename_tag' {
-            request-body -> (:$old!, :$new!) {
-                my $data = $ws.rename-tag($user, $old, $new);
-                content 'application/json', $data;
+            request-body -> (:oldTag($old)!, :newTag($new)!) {
+                my $new-name = $ws.rename-tag($user, $old, $new);
+                content 'application/json', %( :newName($new-name) );
```
Comment (jnthn) [inline lib/Agrammon/Web/Routes.pm6:172]: Also a question of what we do if the name didn't exist, or was in conflict with the new one.

Comment (zaucker) [reply lib/Agrammon/Web/Routes.pm6]: Added error handling and OpenAPI validation

File: lib/Agrammon/DB/Tag.pm6 [hunk 2]
```diff
@@ -22,14 +39,29 @@ class Agrammon::DB::Tag does Agrammon::DB {
...
+            }
+
+            # update failed
+            die X::Agrammon::DB::Tag::RenameFailed.new(:old-name($!name), :new-name($new)) unless $ret.rows;
+
+            # rename suceeded
             $!name = $new;
         }
-        return self;
+        return self.name;
```
Comment (jnthn) [inline lib/Agrammon/DB/Tag.pm6:64]: Is the returned value used anywhere?

File: t/webservice.t [hunk 3]
```diff
@@ -101,17 +101,14 @@ transactionally {
     }
 
     subtest "rename-tag" => {
-        ok my $newTag = $ws.rename-tag($user, '00MyTestTag', '00MyNewTestTag'), "Rename tag";
-        is $newTag.name, '00MyNewTestTag', 'Tag has expected name';
+        ok my $new-name = $ws.rename-tag($user, '00MyTestTag', '00MyNewTestTag'), "Rename tag";
+        is $new-name, '00MyNewTestTag', 'Tag has expected name';
```
Comment (jnthn) [inline t/webservice.t:105]: Well, other than in the test... :-) But still, is it needed.

**Review by jnthn — APPROVED** [review]
One question, but that aside, it looks alright to me.

File: lib/Agrammon/DB/Tag.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/DB/Tag.pm6:64]: Not really ...

File: t/webservice.t (hunk 3 above)
Comment (zaucker) [reply t/webservice.t:105]: ... removed.

**Comment by zaucker** [comment]
Merging after removing unused return value.

**Pull request merged**